libs/util/http.client.js

```diff
@@ -211,6 +211,9 @@ function io(url, options) {
         resp.responseText = body;
 
         if (err) {
+            if (err.message === 'XMLHttpRequest timeout') {
+                err = new Error('XMLHttpRequest timeout: failed to load ' + resp.url + ' in ' + options.timeout + 'ms');
+            }
```

Comment:
> And err.timeout = options.timeout is also set when status code is 0 or 408 (Line 207, a few lines above)

I’ve just looked at the code and I wonder if the current implementation works as you described. As far as I understand the implementation line 207 gets only executed when the upstream error object `err` from `xhr` is falsy, which is only the case for an `418` response but not for a timeout error.